mongo/src/main/java/com/redhat/lightblue/mongo/crud/BasicDocFinder.java

```diff
@@ -96,33 +96,48 @@ public long find(CRUDOperationContext ctx,
                 cursor = cursor.sort(mongoSort);
                 LOGGER.debug("Result set sorted");
             }
-            int numMatched = cursor.count();
-            int nRetrieve=numMatched;
+
+            int numMatched=0;
+            int nRetrieve=0;
+            if(ctx.isComputeMatchCount()) {
```

Comment:
Perhaps we can rename this flag to computeCounts and use it for modifiedCount too? https://github.com/lightblue-platform/lightblue-core/issues/789